### ISSUE TITLE
NaN values in input images

### ISSUE THREAD

**Issue by mpelchat04** [NRCan/geo-deep-learning#68]
## Current behavior
Not a number (NaN) and infinite (inf) values in images are also stored in HDFS files. Those values will affect the loss calculation, resulting in a NaN loss, therefore not learning anything.  
Here's an example of the output (warning from numpy) and training and validation loss are set to nan because of if:  
```shell
Start:

Epoch 0/49
--------------------
/home/ec2-user/miniconda3/lib/python3.6/site-packages/numpy/core/_methods.py:32: RuntimeWarning: invalid value encountered in reduce
  return umr_minimum(a, axis, None, out, keepdims, initial)
Training Loss: nan
Training iou: 0.9478
Training precision: 0.8983
Training recall: 0.9478
Training f1-score: 0.9224
Validation Loss: nan
Validation iou: 0.9395
Validation precision: 0.8827
Validation recall: 0.9395
Validation f1-score: 0.9102
Current elapsed time 27m 17s  
...
```  
## Expected behavior  
NaN values in images should be written as 0 in the hdfs files.  
Infinite values should be reduced.

**Comment by remtav** [comment]
My upcoming pull request makes use of more nan values (e.g. for padding) on sat images, but will convert all these nan values to 0 in the __get_item__ method of Dataset. Is that an acceptable solution? NaN values make it possible to disregard these particular pixels for histogram calculation (e.g. getting the actual value at the 2nd percentile).

**Comment by victorlazio109** [comment]
Resolved
